gguf-py/gguf/gguf_reader.py

```diff
@@ -107,7 +107,13 @@ def __init__(self, path: os.PathLike[str] | str, mode: Literal['r' | 'r+' | 'c']
         offs, tensors_fields = self._build_tensors_fields(offs, tensor_count)
         new_align = self.fields.get('general.alignment')
         if new_align is not None:
-            if new_align.types != [GGUFValueType.UINT64]:
+            if (
+                    (
+                        new_align.types != [GGUFValueType.UINT64]
+                    ) and (
+                        new_align.types != [GGUFValueType.UINT32]
+                    )
+            ):
```

Comment:
This has always been uint32 so this should just be:
```suggestion
            if new_align.types != [GGUFValueType.UINT32]:
```

Reply:
Hmm, I actually have no idea why I used `UINT64` there but I think generally tolerating any integer type should be fine.

So it might seem weird but my response would actually be that just adding `UINT32` doesn't go far enough. Add `UINT8`, `UINT16`, and all the `INT*` types too! @jay-johnson But don't write it as a bunch of `and`s. (Also probably hold off on changes until there's a consensus about what should happen.)

Anyway, what I would do is add some helper static methods to `GGUFValueType` like `is_int`, `is_uint` and then it's convenient to check whether it's some sort of integer value.

I might be inclined to be tolerant everywhere, but in this specific case - a utility to dump GGUF files for informational purposes, I really think it should be as tolerant as possible. Warning about format violations would be fine though. Also this particular field is pretty much always going to fit in even an 8 bit type, it's just the alignment not an absolute offset.

Reply:
If it's [guaranteed](https://github.com/ggerganov/ggml/blob/master/docs/gguf.md#required) by the spec to be uint32, and we have never seen any other type in the wild, why add complexity here?

Reply:
Fair enough, if it's guaranteed in the spec.

@jay-johnson In this case, I recommend just going with cebtenzzre's suggestion.